custom/icds_reports/views.py

```diff
@@ -1718,3 +1721,63 @@ def get(self, request, *args, **kwargs):
             return export_response(sync.get_file_from_blobdb(), 'csv', zip_name)
         except NotFound:
             raise Http404
+
+
+class DishaAPIView(View):
+
+    def message(self, message_name):
+        state_names = ", ".join(self.valid_state_names)
+        error_messages = {
+            "missing_date": "Please specify valid month and year",
+            "invalid_month": "Please specify a month that's older than a month and 5 days",
+            "invalid_state": "Please specify one of {} as state_name".format(state_names),
+            "data_is_unavailable": "No data is available for this month"
+        }
+        return {"message": error_messages[message_name]}
+
+    @method_decorator(api_auth)
+    @json_error
+    def get(self, request, *args, **kwargs):
+        # Todo; add disha API permission
```

Comment:
Will explore a possibility to add a random view that's not a custom report to the `viewable_reports`, so that permission to this API can be given to users using the Webusers/Permissions UI

Reply:
Should this also include some form of domain authentication? like `login_and_domain_required`

Reply:
Made a feature flag